ui/webpack.common.js

```diff
@@ -2,179 +2,198 @@ const path = require("path");
 const TsconfigPathsPlugin = require("tsconfig-paths-webpack-plugin");
 const HtmlWebpackPlugin = require("html-webpack-plugin");
 // webpack 5 stop handling node polyfills by itself, this plugin re-enables the feature
-const NodePolyfillPlugin = require("node-polyfill-webpack-plugin")
+const NodePolyfillPlugin = require("node-polyfill-webpack-plugin");
 const { ModuleFederationPlugin } = require("webpack").container;
 const { federatedModuleName, dependencies } = require("./package.json");
-const ChunkMapper = require('@redhat-cloud-services/frontend-components-config/chunk-mapper')
+const ChunkMapper = require("@redhat-cloud-services/frontend-components-config/chunk-mapper");
 
-
-module.exports = {
-  entry: {
-    app: "./src/index.tsx"
-  },
-  plugins: [
-    new NodePolyfillPlugin(),
-    new HtmlWebpackPlugin({
-      template: "./src/index.html"
-    }),
-    new ChunkMapper({
-      modules: [
-        federatedModuleName
-      ]
-    }),
-    new ModuleFederationPlugin({
-      name: federatedModuleName,
-      filename: "remoteEntry.js",
-      exposes: {
-        "./FederatedArtifactsPage": "./src/app/pages/artifacts/artifacts.federated",
-        "./FederatedArtifactRedirectPage": "./src/app/pages/artifact/artifact.federated",
-        "./FederatedArtifactVersionPage": "./src/app/pages/artifactVersion/artifactVersion.federated",
-        "./FederatedRulesPage": "./src/app/pages/rules/rules.federated"
-      },
-      shared: {
-        ...dependencies,
-        react: {
-          eager: true,
-          singleton: true,
-          requiredVersion: dependencies["react"],
+module.exports = (env, argv) => {
+  const isProduction = argv && argv.mode === "production";
+  return {
+    entry: {
+      app: "./src/index.tsx",
+    },
+    plugins: [
+      new NodePolyfillPlugin(),
+      new HtmlWebpackPlugin({
+        template: "./src/index.html",
+      }),
+      new ChunkMapper({
+        modules: [federatedModuleName],
+      }),
+      new ModuleFederationPlugin({
+        name: federatedModuleName,
+        filename: `${federatedModuleName}${
+          isProduction ? "[chunkhash:8]" : ""
```

Comment:
btw I forgot a dot in this pattern when I did it orginally.
```suggestion
          isProduction ? ".[chunkhash:8]" : ""
```

Reply:
done